app/assets/javascripts/discourse/initializers/copy-codeblocks.js

```diff
@@ -0,0 +1,121 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { later } from "@ember/runloop";
+import { Promise } from "rsvp";
+import { iconHTML } from "discourse-common/lib/icon-library";
+
+// http://github.com/feross/clipboard-copy
+function clipboardCopy(text) {
+  // Use the Async Clipboard API when available. Requires a secure browsing
+  // context (i.e. HTTPS)
+  if (navigator.clipboard) {
+    return navigator.clipboard.writeText(text).catch(function(err) {
+      throw err !== undefined
+        ? err
+        : new DOMException("The request is not allowed", "NotAllowedError");
+    });
+  }
+
+  // ...Otherwise, use document.execCommand() fallback
+
+  // Put the text to copy into a <span>
+  var span = document.createElement("span");
```

Comment:
no var, use let/const, I know it's third party code, but we can't have this in the codebase.